### PR TITLE
Add translation guide + update repo structure

### PR DESCRIPTION
**Pull request by Wauplin**
Triggered by https://github.com/huggingface/huggingface_hub/issues/1601 and [private slack thread](https://huggingface.slack.com/archives/C057VDURL9M/p1689175097996009).

This PR adds:
- a `docs/source/en/` folder in which all current documentation had been moved
- a `docs/README.md` guide to explain how to write docs (adapted from `transformers`)
- a `docs/TRANSLATING.md` guide to explain how to start translating to a new language (adapted from `transformers`)
- a `.github/ISSUE_TEMPLATE/i18n.md` template to create a new translation issue
- a new install group (`pip install -e ".[docs]"`)

The goal is to start translating `huggingface_hub`'s docs to other languages like for `transformers` and `diffusers` libraries.

### CODE REVIEW

**Review by julien-c — COMMENTED** [review]
very supportive of this!

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+You only need to generate the documentation to inspect it locally (if you're planning changes and want to
+check how they look before committing for instance). You don't have to commit the built documentation.
+
+---
+
+## Building the documentation
+
+Once you have setup the `doc-builder` and additional packages, you can generate the documentation by 
+typing the following command:
```
Comment (LysandreJik) [inline docs/README.md:37]: I thought I was missing an instruction at first, but this is included in the instruction above :)
```suggestion
Once you have setup the `doc-builder` and additional packages with the pip install command above, 
you can generate the documentation by typing the following command:
```

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+**NOTE**
+
+The `preview` command only works with existing doc files. When you add a completely new file, you need to update
+`_toctree.yml` & restart `preview` command (`ctrl-c` to stop it & call `doc-builder preview ...` again).
+
+---
+
+## Adding a new element to the navigation bar
+
+Accepted files are Markdown (.md or .mdx).
```
Comment (LysandreJik) [inline docs/README.md:68]: Should we just indicate `.md` and not `.mdx` given that now GitHub renders the mdx files in a non-digestible manner?

**Comment by wonhyeongseo** [comment]
Would love to! We'll start as soon as possible. Thank you for the heads up, @LysandreJik !

Much love from Korea,
Won

File: docs/TRANSLATING.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+**🗞️ Open an issue**
+
+To get started, navigate to the [Issues](https://github.com/huggingface/huggingface_hub/issues) page of this repo and check if anyone else has opened an issue for your language. If not, open a new issue by selecting the "Translation template" from the "New issue" button.
+
+Once an issue exists, post a comment to indicate which chapters you'd like to work on, and we'll add your name to the list.
+
+
+**🍴 Fork the repository**
+
+First, you'll need to [fork the Transformers repo](https://docs.github.com/en/get-started/quickstart/fork-a-repo). You can do this by clicking on the **Fork** button on the top-right corner of this repo's page.
```
Comment (LysandreJik) [inline docs/TRANSLATING.md:14]:
```suggestion
First, you'll need to [fork the `huggingface_hub` repo](https://docs.github.com/en/get-started/quickstart/fork-a-repo). You can do this by clicking on the **Fork** button on the top-right corner of this repo's page.
```

**Review by LysandreJik — APPROVED** [review]
Looks good! Thanks for working on this @Wauplin :raised_hands:

**Review by stevhliu — APPROVED** [review]
Super cool, looking forward to all the translations! 🤗 🌐

File: docs/README.md (hunk 2 above)
Comment (Wauplin) [reply docs/README.md:68]:
```suggestion
Accepted files are Markdown (.md).
```

**Comment by Wauplin** [comment]
Thanks for the review @LysandreJik @stevhliu. I'll merge this PR asap so that future PRs are not impacted by it. I'll work on event/promoting it once I'm done with unpiling my notifications after my time off :)

**Pull request merged**